api/bazel/api_build_system.bzl

```diff
@@ -181,3 +145,50 @@ def api_go_test(name, size, importpath, srcs = [], deps = []):
         importpath = importpath,
         deps = deps,
     )
+
+_GO_BAZEL_RULE_MAPPING = {
+    "@opencensus_proto//opencensus/proto/trace/v1:trace_proto": "@opencensus_proto//opencensus/proto/trace/v1:trace_proto_go",
```

Comment:
I'm less than excited by this one. `api_build_system.bzl` should be generic, but we have per external dependency data here now? Is there anything cleaner or is this unavoidable?

Reply:
Yes, it is inevitable with golang since ODR requires one Go package per proto, and folks generally commit generated code. So there has to be some mapping from bazel proto rules to generated go-code rules (and the package mapping is not at all trivial).

Reply:
OK. I would ideally like to see these moved towards where the third_party deps are actually sourced, e.g. `repositories.bzl`, but for now this is fine.